core/flamingock-test-support/src/main/java/io/flamingock/internal/core/builder/AuditEntryExpectation.java

```diff
@@ -0,0 +1,248 @@
+    private AuditEntry.ExecutionType expectedType;
+    private String expectedClassName;
+    private String expectedMethodName;
+    private Object expectedMetadata;
+    private Long expectedExecutionMillis;
+    private String expectedExecutionHostname;
+    private String expectedErrorTrace;
+    private Boolean expectedSystemChange;
+    private AuditTxType expectedTxType;
+    private String expectedTargetSystemId;
+
+    // Time range for flexible timestamp verification
+    private LocalDateTime timestampAfter;
+    private LocalDateTime timestampBefore;
+
+    // Flags for optional field verification
+    private boolean shouldVerifyExecutionId = false;
+    private boolean shouldVerifyStageId = false;
+    private boolean shouldVerifyTimestamp = false;
+    private boolean shouldVerifyExecutionMillis = false;
+    private boolean shouldVerifyExecutionHostname = false;
+
+    AuditEntryExpectation() {}
+
+    public static AuditEntryExpectation auditEntry() {
+        return new AuditEntryExpectation();
+    }
+
+    public static AuditEntryExpectation STARTED(String taskId) {
+        return new AuditEntryExpectation().withTaskId(taskId).withState(AuditEntry.Status.STARTED);
+    }
+
+    public static AuditEntryExpectation APPLIED(String taskId) {
+        return new AuditEntryExpectation().withTaskId(taskId).withState(AuditEntry.Status.APPLIED);
+    }
+
+    public static AuditEntryExpectation FAILED(String taskId) {
+        return new AuditEntryExpectation().withTaskId(taskId).withState(AuditEntry.Status.FAILED);
+    }
+
+    public static AuditEntryExpectation ROLLED_BACK(String taskId) {
+        return new AuditEntryExpectation().withTaskId(taskId).withState(AuditEntry.Status.ROLLED_BACK);
+    }
+
+    public static AuditEntryExpectation ROLLBACK_FAILED(String taskId) {
+        return new AuditEntryExpectation().withTaskId(taskId).withState(AuditEntry.Status.ROLLBACK_FAILED);
+    }
+
+    @Deprecated
+    public static AuditEntryExpectation started(String changeId) {
+        return STARTED(changeId);
+    }
+
+    @Deprecated
+    public static AuditEntryExpectation applied(String changeId) {
+        return APPLIED(changeId);
+    }
+
+    @Deprecated
+    public static AuditEntryExpectation failed(String changeId) {
+        return FAILED(changeId);
+    }
+
+    @Deprecated
+    public static AuditEntryExpectation rolledBack(String changeId) {
+        return ROLLED_BACK(changeId);
+    }
+
+    // Identity fields
+    public AuditEntryExpectation withExecutionId(String executionId) {
+        this.expectedExecutionId = executionId;
+        this.shouldVerifyExecutionId = true;
+        return this;
+    }
+
+    public AuditEntryExpectation withStageId(String stageId) {
+        this.expectedStageId = stageId;
+        this.shouldVerifyStageId = true;
+        return this;
+    }
+
+    public AuditEntryExpectation withTaskId(String taskId) {
+        this.expectedTaskId = taskId;
+        return this;
+    }
```

Comment:
We can remove this because it should always be created with static constructors(`APPLIED`, `FAILED`, etc)